pkg/operators/otel-metrics/otel-metrics.go

```diff
@@ -0,0 +1,485 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelmetrics
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	name = "otel-metrics"
+
+	Priority                      = 50000
+	ParamOtelMetricsEnabled       = "otel-metrics-enabled"
+	ParamOtelMetricsListenAddress = "otel-metrics-listen-address"
+
+	MetricTypeKey       = "key"
+	MetricTypeCounter   = "counter"
+	MetricTypeGauge     = "gauge"
+	MetricTypeHistogram = "histogram"
+)
+
+type otelMetricsOperator struct {
+	exporter      *prometheus.Exporter
+	meterProvider metric.MeterProvider
+	initialized   bool
+	skipListen    bool
+}
+
+func (m *otelMetricsOperator) Name() string {
+	return name
+}
+
+func (m *otelMetricsOperator) Init(globalParams *params.Params) error {
+	if m.initialized {
+		return nil
+	}
```

Comment:
Actually, thinking about it again: this would unnecessarily complicate the code - we would need to store errors and return them on subsequent calls. #2901 moves from lazily instantiating operators towards a clean solution, which prevents operators from being initialized more than once. So I think after that is merged, we can keep it more straightforward like it is right now.